src/lifecycle/properties-init.js

```diff
@@ -23,22 +28,30 @@ function createNativePropertyDefinition (name, opts) {
 
     if (typeof opts.default === 'function') {
       info.defaultValue = opts.default(elem);
-    } else if (opts.default !== undefined) {
+    } else if (!empty(opts.default)) {
       info.defaultValue = opts.default;
     }
+    
+    const defaultValue = info.defaultValue;
+    const defaultValueIsEmpty = empty(defaultValue);
 
-    // TODO Refactor
+    // TODO Refactor to be cleaner.
     if (info.linkedAttribute) {
       if (!info.attributeMap) {
         info.attributeMap = {};
 
         elem.removeAttribute = function (attrName) {
           info.updatingAttribute = true;
-          info.removeAttribute.call(this, attrName);
+
+          if (defaultValueIsEmpty) {
```

Comment:
This ensures that if a default value exists, that the attribute is returned to that value so it is always in sync with the prop.